File: apps/web/src/types/quote.ts
export type QuoteStatus = "PENDIENTE" | "APROBADA" | "RECHAZADA";
export type QuoteItemType = "MANO_OBRA" | "PIEZA";

export interface QuoteItem {
  id: string;
  item_type: QuoteItemType;
  description: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
}

export interface Quote {
  id: string;
  work_order_id: string;
  status: QuoteStatus;
  apply_itbis: boolean;
  itbis_rate: number;
  subtotal: number;
  itbis_amount: number;
  total: number;
  notes: string | null;
  items: QuoteItem[];
  created_at: string;
  updated_at: string;
}

export interface QuoteItemInput {
  itemType: QuoteItemType;
  description: string;
  quantity: number;
  unitPrice: number;
}

export interface CreateQuoteRequest {
  workOrderId: string;
  applyItbis: boolean;
  notes?: string;
  items: QuoteItemInput[];
}

File: apps/api/src/main/java/rd/tallerfacil/api/quote/domain/Quote.java
package rd.tallerfacil.api.quote.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import rd.tallerfacil.api.shared.domain.BaseEntity;
import rd.tallerfacil.api.workorder.domain.WorkOrder;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "quotes")
@Getter
@Setter
@NoArgsConstructor
public class Quote extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "work_order_id", nullable = false)
    private WorkOrder workOrder;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private QuoteStatus status = QuoteStatus.PENDIENTE;

    @Column(name = "apply_itbis", nullable = false)
    private boolean applyItbis = false;

    @Column(name = "itbis_rate", nullable = false, precision = 5, scale = 4)
    private BigDecimal itbisRate = new BigDecimal("0.18");

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal subtotal = BigDecimal.ZERO;

    @Column(name = "itbis_amount", nullable = false, precision = 12, scale = 2)
    private BigDecimal itbisAmount = BigDecimal.ZERO;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal total = BigDecimal.ZERO;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @OneToMany(mappedBy = "quote", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<QuoteItem> items = new ArrayList<>();
}
